astropy/io/fits/fitsrec.py

```diff
@@ -553,11 +550,13 @@ def __setitem__(self, key, value):
             raise TypeError('Assignment requires a FITS_record, tuple, or '
                             'list as input.')
 
-    def __getslice__(self, start, end):
-        return self[slice(start, end)]
+    if PY2:
```

Comment:
@eric-wieser - in principle, you're right, but since this only makes an unnecessary define, it doesn't matter too much.  Also, the whole clause is going to be removed in June, since we'll drop python 2 support then (though this will be in our 2.0 long-term support release). By marking it `if PY2`, we're sure to find it...